consensus/polybft/fsm.go

```diff
@@ -161,8 +162,33 @@ func (f *fsm) BuildProposal() (*pbft.Proposal, error) {
 	}
 
 	// fill the block with transactions
-	if err := f.blockBuilder.Fill(); err != nil {
-		return nil, err
+	f.blockBuilder.Fill()
+
+	receipts := f.blockBuilder.Receipts()
+	events := make([]*ExitEvent, 0)
+
+	for i := 0; i < len(receipts); i++ {
+		if len(receipts[i].Logs) == 0 {
+			continue
+		}
+
+		event, err := decodeExitEvent(convert(receipts[i].Logs[0]), f.epochNumber, parent.Number+1)
+		if err != nil {
+			if errors.Is(err, errNotAnExitEvent) {
+				// valid case, transaction is not an exit transaction
+				continue
+			}
```

Comment:
Nitpick. Why not call errors.Is(err, errNotAnExitEvent) before nil check? You can write shallower code.

Reply:
I removed that error completely. If the log does not contain an exit event (which is a valid case), that `decode` will return `nil, nil`, as per comment: https://github.com/0xPolygon/polygon-edge/pull/852#discussion_r1009656522

Fix: https://github.com/0xPolygon/polygon-edge/pull/852/commits/d9bbc65d412e67074c623708a184a265a2fbbd4c